client/trino-client/src/main/java/io/trino/client/ResultRowsDecoder.java

```diff
@@ -165,4 +185,20 @@ private static ResultRows concat(List<ResultRows> resultRows)
                 .collect(toImmutableList())
                 .iterator());
     }
+
+    private static void waitForMemory(long requiredMemory)
+    {
+        MemoryMXBean memoryBean = ManagementFactory.getMemoryMXBean();
```

Comment:
This is racy. Instead of going up to the Xmx, this should be a configurable property that is passed to the JDBC driver, let's call it `prefetchingBufferSize`